notification/rule/slack.go

```diff
@@ -13,6 +18,129 @@ type Slack struct {
 	MessageTemplate string `json:"messageTemplate"`
 }
 
+// GenerateFlux generates a flux script for the slack notification rule.
```

Comment:
Seems like this shares a lot with the other generation code. Does it seem appropriate to remove the repetition?

Reply:
Definitely want to get rid of the repetition. Was in a hurry trying to get the demo here straight.

There's definitely mixed concepts in this PR. It really should have been two PRs. One for generating the notification rule task and the other for generating the flux for a notification rule.

Reply:
@chnn for now I'm going to hold off on reducing any of the repetition. Opened https://github.com/influxdata/influxdb/issues/14702